File: 2762-continuous-subarrays/2762-continuous-subarrays.cpp
class Solution {
public:
    long long clac(int l,int r){
        long long n = r - l;
        return n*(n+1)/2;
    }
    long long continuousSubarrays(vector<int>& nums) {
        long long sum = 0;
        int l = 0 , r = 0 , n = nums.size();
        int mn = nums[0],mx=nums[0];
        while(r<n){
            mn = min(mn,nums[r]);
            mx = max(mx,nums[r]);
            if(mx - mn > 2){
                sum+=clac(l,r); // 3
                l = r;
                mn = mx = nums[r];
                while(l && abs(nums[r] - nums[l-1]) <= 2){
                    l--;
                    mn = min(mn,nums[l]);
                    mx = max(mx,nums[l]);
                }
                sum-=clac(l,r);
            }
            r++;
        }
        return sum+clac(l,r);
    }
};

File: 2762-continuous-subarrays/2762-continuous-subarrays.py
class Solution:
    def calc(self, l: int, r: int) -> int:
        n = r - l
        return n * (n + 1) // 2

    def continuousSubarrays(self, nums: list[int]) -> int:
        total_sum = 0
        l, r = 0, 0
        n = len(nums)
        mn, mx = nums[0], nums[0]

        while r < n:
            mn = min(mn, nums[r])
            mx = max(mx, nums[r])

            if mx - mn > 2:
                total_sum += self.calc(l, r)
                l = r
                mn = mx = nums[r]
                while l > 0 and abs(nums[r] - nums[l - 1]) <= 2:
                    l -= 1
                    mn = min(mn, nums[l])
                    mx = max(mx, nums[l])
                total_sum -= self.calc(l, r)

            r += 1

        return total_sum + self.calc(l, r)
